### PR TITLE
PV2-1498 - Speed up Monitoring of Large Providers

### PR DESCRIPTION
**Pull request by benagain**
Streamline the monitoring activity when processing a large provider through the Payments v2 monitoring service by creating specific nodes for the top 5 providers.

### CODE REVIEW

File: src/SFA.DAS.Payments.Monitoring.Jobs.Client/EarningsJobClient.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ public class EarningsJobClient : IEarningsJobClient
         private readonly IMessageSession messageSession;
         private readonly IPaymentLogger logger;
         private readonly IConfigurationHelper config;
+        private readonly JobMonitorPartition partitionName = new JobMonitorPartition();
```
Comment (carlbm) [inline src/SFA.DAS.Payments.Monitoring.Jobs.Client/EarningsJobClient.cs]: Can you pass this in as a constructor argument instead? Would just make it more testable

File: src/SFA.DAS.Payments.Monitoring.Jobs.Client.UnitTests/PartitionNameTests.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        [TestCase(72, 0, 12)]
+        [TestCase(73, 0, 13)]
+        [TestCase(74, 0, 14)]
+        [TestCase(75, 0, 15)]
+        [TestCase(76, 0, 16)]
+        [TestCase(77, 0, 17)]
+        [TestCase(78, 0, 18)]
+        [TestCase(79, 0, 19)]
+        [TestCase(80, 0, 0)]
+        [TestCase(1, 10003915, 10003915)]
```
Comment (carlbm) [inline src/SFA.DAS.Payments.Monitoring.Jobs.Client.UnitTests/PartitionNameTests.cs]: Can you use a different job id for the same UKPRN? Just to prove that this isn't a happy mistake

File: src/SFA.DAS.Payments.Monitoring.Jobs.Client/JobMonitorPartition.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿using System.Linq;
+
+namespace SFA.DAS.Payments.Monitoring.Jobs.Client
+{
+    public class JobMonitorPartition
+    {
+        public static long[] InvidivualNodes = new long[]
```
Comment (carlbm) [inline src/SFA.DAS.Payments.Monitoring.Jobs.Client/JobMonitorPartition.cs]: Use a hashset instead? Really not 100% though :)

File: src/SFA.DAS.Payments.Monitoring.Jobs.Client/EarningsJobClient.cs (hunk 1 above)
Comment (benagain) [reply src/SFA.DAS.Payments.Monitoring.Jobs.Client/EarningsJobClient.cs]: This is a good idea, when the tests are written :)

File: src/SFA.DAS.Payments.Monitoring.Jobs.Client/JobMonitorPartition.cs (hunk 3 above)
Comment (benagain) [reply src/SFA.DAS.Payments.Monitoring.Jobs.Client/JobMonitorPartition.cs]: I'm not sure it's worth it for a collection of 5 values?

File: src/SFA.DAS.Payments.Monitoring.Jobs.Client.UnitTests/PartitionNameTests.cs (hunk 2 above)
Comment (benagain) [reply src/SFA.DAS.Payments.Monitoring.Jobs.Client.UnitTests/PartitionNameTests.cs]: Good idea, done 👍

**Pull request closed without merging**